### PR TITLE
doc/ Update Exchange Connector developer documentation

### PR DESCRIPTION
**Pull request by CrimsonJacket**

### CODE REVIEW

File: content/developer/overview.mdx [hunk 1]
```diff
@@ -61,12 +69,12 @@ We encourage and welcome contributions from the community, subject to the guidel
 - The best way to create a connector that adheres to Hummingbot’s standard is by cloning the logic of existing connectors. We’ve done a lot of work to build our core connectors, so no need to reinvent the wheel.
```
Comment (Nullably) [inline content/developer/overview.mdx:69]: I would add this as well

- Make sure every REST API call is done in asynchronous fashion.

File: content/developer/task1.mdx [hunk 2]
```diff
@@ -63,165 +111,138 @@ Fetches or Subscribes to the order book deltas(diffs) for each trading pair. Add
...
 
-Get all asks on the order book in dictionary format. <br/>
+The following details are **required** functions to be implemented in the new `OrderBook` class:
 
-**Input Parameter:** None <br/>
-**Expected Output(s):** `Dict[Decimal, Dict[str, Dict[str, any]]]`
+### `snapshot_message_from_exchange`
 
-### `active_bids`
+Converts json snapshot data from the exchangee into standard `OrderBookMessage` format.
```
Comment (Nullably) [inline content/developer/task1.mdx:131]: typo, exchangee

File: content/developer/task1.mdx [hunk 3]
```diff
@@ -63,165 +111,138 @@ Fetches or Subscribes to the order book deltas(diffs) for each trading pair. Add
...
 
-Get all bids on the order book in dictionary format. <br/>
+**Input Parameter(s):** msg: `Dict[str, any]`, timestamp: `float`, metadata: `Optional[Dict] = None` <br/>
+**Expected Output(s):** `OrderBookMessage`
 
-**Input Parameter:** None <br/>
-**Expected Output(s):** `Dict[Decimal, Dict[str, Dict[str, any]]]`
+### `snapshot_message_from_db`
```
Comment (Nullably) [inline content/developer/task1.mdx:136]: I don't think we should mention this or anything that is not typically required.

File: content/developer/task1.mdx [hunk 4]
```diff
@@ -63,165 +111,138 @@ Fetches or Subscribes to the order book deltas(diffs) for each trading pair. Add
...
-### `convert_diff_message_to_order_book_row`
+Converts json diff data from the exchange into standard `OrderBookMessage` format.
 
-Convert an incoming diff message to Tuple of `np.arrays`, and then convert to `OrderBookRow`. <br/>
+**Input Parameter(s):** msg: `Dict[str, any]`, timestamp: `float`, metadata: `Optional[Dict] = None` <br/>
+**Expected Output(s):** `OrderBookMessage`
 
-**Input Parameter:** `object`: message <br/>
-**Expected Output(s):** `Tuple[List[OrderBookRow],List[OrderBookRow]]`
+### `diff_message_from_db`
```
Comment (Nullably) [inline content/developer/task1.mdx:150]: same, no need to have this in the doc.

File: content/developer/task1.mdx [hunk 5]
```diff
@@ -63,165 +111,138 @@ Fetches or Subscribes to the order book deltas(diffs) for each trading pair. Add
...
-Convert an incoming trade message to Tuple of `np.arrays`, and then convert to `OrderBookRow`. <br/>
+**Input Parameter(s):** record: `RowProxy`, metadata: `Optional[Dict] = None` <br/>
+**Expected Output(s):** `OrderBookMessage`
 
-**Input Parameter:** `object`: message <br/>
-**Expected Output(s):** `Tuple[List[OrderBookRow],List[OrderBookRow]]`
+### `trade_message_from_exchange`
 
-### `c_convert_snapshot_message_to_np_arrays`
+Converts json trade data from the exchangee into standard `OrderBookMessage` format.
```
Comment (Nullably) [inline content/developer/task1.mdx:159]: typo

File: content/developer/task3.mdx [hunk 6]
```diff
@@ -1,444 +1,560 @@
...
-- Listening on user stream<br/>
-  This is primarily done in the `_user_stream_event_listener` function. This is only done when the exchange has a WebSocket API.
+### **_async_** `_create_order()`
 
-- Periodic status polling<br/>
-  This serves as a fallback for when user stream messages are not caught by the `UserStreamTracker`. This is done by the `_status_polling_loop`
+This function is responsible for executing the API request to place the order on the exchange. It does the following:
 
-The table below details the **required** functions to implement
+- Verifies that the orders would be legal based on the trading rules pulled from the exchange.
```
Comment (Nullably) [inline content/developer/task3.mdx:125]: ~~Verifies that the orders would be legal based on the trading rules pulled from the exchange.~~
Verifies that the order satisfies exchange trading rules.

**Review by Nullably — CHANGES_REQUESTED** [review]
Well written documentation. 
Thanks a lot @CrimsonJacket  for taking the time to do this.
Some minor requested changes.

**Comment by CrimsonJacket** [comment]
@Nullably Addressed your comments and made additional revisions. If it is sufficient, lets push it to `master` sooner rather than later. Seems like there's some feedback on the `#dev-*` channels wrt to updating our developer docs.

**Review by Nullably — APPROVED** [review]
LGTM

**Comment by dennisocana** [comment]
Resolved a conflict due to a recent change in `Callout.js` and merged the PR.

**Pull request merged**